Makefile

```diff
@@ -1,51 +1,127 @@
-GOCMD ?= go
-GORUN ?= $(GOCMD) run
-GOBUILD ?= $(GOCMD) build
-GOBUILDFLAGS ?= -mod=vendor -ldflags "-X k8s.io/client-go/pkg/version.gitCommit=$$(git rev-parse HEAD) -X k8s.io/client-go/pkg/version.gitVersion=v1.0.0+$$(git rev-parse --short=7 HEAD)"
-GOBUILDDEBUGFLAGS ?= -gcflags="all=-N -l"
-GOTEST ?= $(GOCMD) test
-GOGET ?= $(GOCMD) get
-GOMOD ?= $(GOCMD) mod
-GOFMT ?= gofmt
-GOLINT ?= golint
-CONTAINER_RUNTIME ?= podman
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# Testing
+GO_TEST_FLAGS = $(VERBOSE)
+COVER_PROFILE = cover.out
+
+# Build
+CLIENTGO_VERSION := $(shell git rev-parse --short=7 HEAD)
+CLIENTGO_COMMIT := "v1.0.0+$(shell git rev-parse HEAD)"
+export LDFLAGS="-X k8s.io/client-go/pkg/version.gitCommit=${CLIENTGO_COMMIT} \
+			-X k8s.io/client-go/pkg/version.gitVersion=${CLIENTGO_VERSION}"
+
+# Configuration
+KUBECONFIG ?= config/local.yaml
+RUN_FLAGS ?= -v4
+
+# Tools
+CONTAINER_RUNTIME := $(shell command -v podman 2> /dev/null || echo docker)
+GOLANGCI_LINT := $(GOBIN)/golangci-lint
+
+export GO111MODULE=on
+export GOFLAGS=-mod=vendor
+
+## --------------------------------------
+## Tests
+## --------------------------------------
+
+# Run the tests
+.PHONY: test
+test: unit
+
+# Run the unit tests
+.PHONY: unit
+unit:
+	go test ./... $(VERBOSE) -coverprofile $(COVER_PROFILE)
+
+.PHONY: unit-cover
+unit-cover:
+	go test -coverprofile=$(COVER_PROFILE) $(GO_TEST_FLAGS) ./...
+	go tool cover -func=$(COVER_PROFILE)
+
+.PHONE: unit-verbose
+unit-verbose:
+	VERBOSE=-v make unit
+
+## --------------------------------------
+## Linting
+## --------------------------------------
+
+.PHONY: sec
+sec: lint
+
+.PHONY: fmt
+fmt: lint
+
+.PHONY: vet
+vet: lint
```

Comment:
Do we need all these commands? Every command calls the linting. Also the `vet` command was previously really vetting the code. Is that part of the lint run?

Reply:
Yeh, I was also thinking about it... ´Cause I got the inspiration from [baremetal-operator](https://github.com/openshift/baremetal-operator/blob/master/Makefile) and they have those commands. But I´m also questioning its value. If you agree, and nobody sees any value to have it, we can remove it for sure.

I´ll definilly remote them.